### DISCUSSION TITLE
How to rename a vhost

### DISCUSSION THREAD

**Discussion by hatharom** [rabbitmq/rabbitmq-server#7818] [Other]
### Describe the bug

It is more of a question

I have a running RabbitMQ with vhost named "app1" and it already contains exchanges,queues and messages.

Now I would like to rename it to something different lets say "app2"

Is there a way to do that?

Like
- Renaming it
- Clone it(with topology,messages,permissions) to a new vhost called app2 then delete the old one

Couldn't find any solution for the above

### Reproduction steps

-

### Expected behavior

-

**Comment by michaelklishin** [comment]
Please use Discussions and not Issues for questions.

There is no way to rename a virtual host. You can [export definitions](https://rabbitmq.com/definitions.html), rename it in the JSON file and import it back in,
then delete the original host after all applications were updated to use the new one.

**Comment by michaelklishin** [comment]
Oh, and that won't move messages. You can move them from virtual host A to B using [Dynamic shovels](https://rabbitmq.com/shovel.html).
